components/ai_chat/resources/page/components/markdown_renderer/index.tsx

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Markdown from 'react-markdown'
+
+import styles from './style.module.scss'
+import CaretSVG from '../svg/caret'
+
+const CodeBlock = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Block
+}))
+const CodeInline = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Inline
+}))
+
+const allowedElements = [
+  'blockquote',
+  'br',
+  'code',
+  'del',
+  'em',
+  'h1',
+  'h2',
+  'h3',
+  'h4',
+  'h5',
+  'h6',
+  'hr',
+  'li',
+  'li',
+  'ol',
+  'ol',
```

Comment:
Duplicates.

Reply:
Might even consider a slightly different layout in the code, for easier maintenance:

```js
const allowedElements = [
  // Headings
  'h1', 'h2', 'h3', 'h4', 'h5', 'h6',

  // Text formatting
  'blockquote', 'code', 'del', 'em', 'strong', 'sup',

  // Lists
  'li', 'ol', 'ul',

  // Structural elements
  'p', 'pre', 'section',

  // Line elements
  'br', 'hr'
];
```

Reply:
removed dupes